crates/cubecl-linalg/src/tensor/identity.rs

```diff
@@ -0,0 +1,79 @@
+use cubecl::frontend::TensorHandleRef;
+use cubecl::prelude::*;
+use cubecl::{calculate_cube_count_elemwise, tensor_line_size_parallel};
+use cubecl_core as cubecl;
+
+use super::TensorHandle;
+
+#[cube(launch_unchecked)]
+fn identity_kernel<C: Numeric>(output: &mut Tensor<Line<C>>, gap: u32) {
+    if ABSOLUTE_POS < output.len() {
+        let mut line = Line::empty(output.line_size()).fill(C::from_int(0));
+
+        let start_pos = ABSOLUTE_POS * output.line_size();
+        let mut offset = 0;
+        while offset < output.line_size() {
+            let remainder = (start_pos + offset) % gap;
+            if remainder % gap == 0 {
+                line[offset] = C::from_int(1);
+                offset += gap;
+            } else {
+                offset += gap - remainder;
+            }
+        }
+        output[ABSOLUTE_POS] = line;
+    }
+}
+
+/// Launch identity matrix kernel.
+/// Ensure output is a tensorhandle containing a square matrix with shape 2 x 2.
+/// output will contain the identity matrix.
```

Comment:
The shape comment is wrong I think, it's not always 2x2?

Reply:
Solved